modules/application/src/main/java/edu/northeastern/truthtree/adapter/basicInfo/BasicInfoDBAdapter.java

```diff
@@ -199,8 +266,8 @@ private JSONArray addPopulationToBasicInfo(Map basicInfoData, JSONArray response
    * @return JSONArray representing the data on CITIES_URL
    */
   @Override
-  public JSONArray getBasicCitiesInfo() {
-    return JSONUtil.readJSONFile(CITIES_FILE_PATH);
+  public PageDTO<LocationDTO> getBasicCitiesInfo() {
+    return jsonArrayToLocationPage(JSONUtil.readJSONFile(CITIES_FILE_PATH));
```

Comment:
You can directly used one-time initialized cities values from static block. It will avoid reading from file in every request.

Reply:
addressed